src/devices/src/virtio/block/device.rs

```diff
@@ -323,16 +359,14 @@ impl Block {
             });
 
             if queue.prepare_kick(mem) {
-                self.irq_trigger
-                    .trigger_irq(IrqType::Vring)
-                    .unwrap_or_else(|_| {
-                        METRICS.block.event_fails.inc();
-                    });
+                self.irq_trigger.trigger_vring_irq(batch_irqs);
```

Comment:
To keep the code simple, could we "delay" the IRQ here for both interrupt suppression and no-suppression cases?

Basically "batch" IRQs in both cases? (might introduce extra latency for suppression case, but not sure if that's all or if it messes with the mechanism in other ways)

Reply:
If we can't delay, then I suggest we call `enqueue_irq()` or `trigger_irq()` here based on `batch_irqs` flag to keep logic clear and centered:

```suggestion
                if batch_irqs {
                    self.irq_trigger.enqueue_vring_irq();
                } else {
                    self.irq_trigger.trigger_vring_irq();
                }
```

Reply:
As we discussed offline we decided to drop batching for both cases (with and wo notif suppression). This will help us keep the code simple.

Since all kernels >=4.14 support notification suppression, there are only few corner cases where the performance regression would show:
- very old or custom kernel that doesn't support notif suppression
- new kernel with notif suppression disabled (for snapshot compatibility for example)

So for the moment we'll drop batching and revisit it later if needed.